java/arcs/android/sdk/host/IntentArcHostAdapter.kt

```diff
@@ -86,7 +85,7 @@ class IntentArcHostAdapter(
 
     private class ResultReceiverStateChangeHandler(
         val block: (ArcId, ArcState) -> Unit
-    ) : ResultReceiver(Handler()) {
+    ) : ResultReceiver(null) {
```

Comment:
What is best practice in Android client code for this -- here (and below in IntentHostAdapter) we change to run on an [arbitrary thread](https://developer.android.com/reference/android/os/ResultReceiver#ResultReceiver(android.os.Handler)). Are these two classes used in client code currently, or are we just being defensive here as part of the rewrite to minimize multithreading generally?

Reply:
It was sort-of arbitrary before. Invoking Handler() would use the default Looper of the thread that invoked it.  However, Some threads don't have loopers, and so it throws in exception. If you happened to be calling Allocator.startArc() from  the Main thread, it would work, but if you called startArc() from some other thread, sometimes it would throw an exception about the lack of a looper. Passing a null there basically makes the Allocator use an extra thread, but the looper is prepared for it. 

I'll try changing this to (Handler(myLooper)) and have myLooper call prepare() if it hasn't be invoked and see if that works. But I don't know the downside of installing a Looper onto the caller's thread. 

@ianull do you know?

Reply:
Also, this code isn't used in any of the current integrations, it only gets used if you have multi-hosts across services/processes.

Reply:
ResultReceiver(null) would execute `send()` synchronously at the calling thread instead of posting a message to another thread a `Looper` was installed. In our context, the `onReceiveResult()` will be executed synchronously when calling `send()` without waking up other threads nor suspending itself. So the null here is safe imo.

A `Thread` can allow one single `Looper` installed, the 2nd and subsequent `prepare()` would throw an exception immediately. `prepare()` needs to pair with `loop()` to enter the event loop and never leaves until `quit()` is called or the thread is killed. Calling `prepare()` without `loop()` takes no effect and will not react to any posted messages and runnables iirc.

Reply:
The null is safe in a single process environment, but not if its an IPC across processes tho? Because the caller of send() could be in another process. 

I changed this to use Handler(Looper.mainLooper()). This runs the onReceiveResult immediately on the main looper, which then calls continuation.resume() which will immediately yield back the looper and perform the actual work on the coroutine thread. At least, when I tested this on Kotlin Playground, continuation.resume() seems to resume on the original thread. 

I'm open to other suggestions. Remember, this code isn't going to be used right now, but is for multiprocess isolation in the future.